### DISCUSSION TITLE
Melee requires QOL changes and monster melee rebalancing

### DISCUSSION THREAD

**Discussion by Kaillera** [FakeFishGames/Barotrauma#9482] [Ideas and suggestions]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

So I'm usually playing submarines that are susceptible to aliens breaching in, and I'm seeing lots of problems that only ever increased ever since stun batons got nerfed and the autofill/economy rebalancing.

Aliens have became too strong in melee combat.
- Players are stunned instantly when aliens attacks us, and it can last long.
- They can inflict concussion on the players which makes it even worst for a downhill fight.
- It is impossible to engage in a team fight. You and your friends or bots will generally friendly fire eachother in melee combat, while aliens don't have any friendly fire issues.
- Aliens recover health from hurting live characters, not from just eating. (This is especially unbalanced when you go up against a single husk crawler).
- Alien attacks are instant, you don't see the attack. It just happens as long as you're right next to them.

I suggest some of the following:
- When human characters melee, the attack prioritizes hitting living enemies, and ignore corpses unless it's literally the only target left. Otherwise friendly fire allies. Currently, even if friendly fire is off, you still waste the action hitting eachother.
- Remove enemy life leech from attacks. They already gain health from eating. Try soloing a husk crawler while swimming.
- Reduce the stun time from Aliens.
- Give Aliens some sort of proper attack animation, taunt or quirk animation moments before they attack or at some various intervals. This will give players an opportunity to dodge or attack, rather than trying to compete for the "whoever hits first wins". Aliens knows exactly when to hit you, and they attack instantaneously too, which is usually in their favour.

Some fun ideas are crawler doing a summersault then a dive when attacking the sub outside, giving the players small opportunity to hit them with turrets. When crawlers are inside, they do a jumping laughing taunt. For mudraptors, they would roar at the player before charging inside a sub, or dive attacking the sub.

### Reproduction steps

1. Fight 1 vs 1 in melee with any aliens other than a standard crawler.
2. Fight with multiple AI on fight intruders order against two or more aliens all equipped with melee.

### Bug prevalence

Happens every time I play

### Version

0.18.12.0

### Which operating system did you encounter this bug on?

Windows

**Comment by Ek-Videogames** [comment]
To long-story-short what has already been said to exhausting detail on this subject in the discord & forums:
Nerfing of stun batons has made 'boring-playstylers' come face to face with the realities of monster & human melee stats.

I agree that more enemy attack 'windows' during which contact with a monster does not result in immediate death is a fair concept for compromise rather than buffing available melee options directly. Also agree with better hitbox reliability.

**Comment by Kaillera** [comment]
Got some video from Mudraptor Gaming in Discord to show how much of an issue the monster stuns really is:

https://cdn.discordapp.com/attachments/220881718666723328/999008675995852930/2022-07-19_13-40-59.mp4
https://cdn.discordapp.com/attachments/220881718666723328/999012294157664356/2022-07-19_13-49-28.mp4
https://cdn.discordapp.com/attachments/220881718666723328/999012294543556708/2022-07-19_13-54-28.mp4
